cmd/vicadmin/vicadm.go

```diff
@@ -54,7 +48,33 @@ import (
 	"github.com/vmware/vic/pkg/vsphere/vm"
 )
 
+/*
+    // Note: Including CGo routines here for setuid and setgid,
+    // Because the Setuid and Setgid functions are actually disabled
+    // (See issue here: https://github.com/golang/go/issues/1435)
+
+    // The key is that we're calling setuid/setgid() before spawning
+    // ANY goroutines, therefore all goroutines should have the expected
+    // UID/GID
+
+#include <errno.h>
+#include <string.h>
+#include <sys/types.h>
+#include <unistd.h>
+int do_setuid(int id) {
+	return setuid(id);
+}
+int do_setgid(int id) {
+	return setgid(id);
+}
+char* do_strerror() {
+	return strerror(errno);
+}
+*/
+import "C"
```

Comment:
It's still not clear how this is going to work. This is only setting cgo and not the go runtime. The issue you referred here https://github.com/golang/go/issues/1435 suggests using https://github.com/golang/go/issues/1435#issuecomment-173804075 so tether should do that instead of vicadmin trying to call setuid/gid